### DISCUSSION TITLE
How to build a community container for paperless-ngx

### DISCUSSION THREAD

**Discussion by michaelfliegner** [nextcloud/all-in-one#5641] [Ideas] [answered]
Could s.o. provide some instructions to jump start a docker newbie with that?

**Comment by szaimen** [comment, marked as answer]
Hey, so the trick is to translate the given docker-compose like e.g. this one https://github.com/paperless-ngx/paperless-ngx/blob/dev/docker/compose/docker-compose.sqlite.yml to the json syntax used in AIO like stated in https://github.com/nextcloud/all-in-one/tree/main/community-containers#how-to-add-containers:

```json
{
    "aio_services_v1": [
        {
            "container_name": "nextcloud-aio-paperless-ngx",
            "display_name": "Paperless-ngx",
            "documentation": "https://github.com/nextcloud/all-in-one/tree/main/community-containers/paperless-ngx",
            "depends_on": [
              "nextcloud-aio-paperless-ngx-redis"
            ],
            "image": "paperlessngx/paperless-ngx",
            "image_tag": "latest",
            "internal_port": "8000",
            "restart": "unless-stopped",
            "ports": [
                {
                  "ip_binding": "%APACHE_IP_BINDING%",
                  "port_number": "8000",
                  "protocol": "tcp"
                }
            ],
            "environment": [
                "TZ=%TIMEZONE%",
                "COMPOSE_PROJECT_NAME=paperless",
                "PAPERLESS_REDIS=redis://nextcloud-aio-paperless-ngx-redis:6379"
            ],
            "volumes": [
                {
                    "source": "nextcloud_aio_paperless_ngx_data",
                    "destination": "/usr/src/paperless/data",
                    "writeable": true
                },
                {
                    "source": "nextcloud_aio_paperless_ngx_media", // or should the nextcloud datadir be mounted here? What is the media folder for?
                    "destination": "/usr/src/paperless/media",
                    "writeable": true
                }/* Not 100% sure on the volumes below:,
                {
                    "source": "nextcloud_aio_paperless_ngx_export", // should this be included in the backup solution?
                    "destination": "/usr/src/paperless/export",
                    "writeable": true
                },
                {
                    "source": "nextcloud_aio_paperless_ngx_consume", // should this be Nextclouds datadir or can it be ommitted?
                    "destination": "/usr/src/paperless/consume",
                    "writeable": true // does this needs to be writeable?
                }*/
            ],
            "backup_volumes": [
                "nextcloud_aio_paperless_ngx_data" // and more?
            ]
        },
        {
            "container_name": "nextcloud-aio-paperless-ngx-redis",
            "display_name": "Paperless-ngx-redis",
            "documentation": "https://github.com/nextcloud/all-in-one/tree/main/community-containers/paperless-ngx",
            "image": "redis",
            "image_tag": "7",
            "internal_port": "6379",
            "restart": "unless-stopped",
            "environment": [
                "TZ=%TIMEZONE%"
            ],
            "volumes": [
                {
                    "source": "nextcloud_aio_paperless_ngx_redis",
                    "destination": "/data",
                    "writeable": true
                }
            ]
        }
    ]
}
```

If you want to step in as the maintainer, please create a PR by following https://github.com/nextcloud/all-in-one/tree/main/community-containers#how-to-add-containers. Let me know if you need any help on this :)

**Comment by caplam** [comment]
Interesting. I discover the community containers.
I installed aio long time ago on an unraid server so the path of aio folders is a bit weird compared to other unraid docker containers.
I have a running instance of paperless-ngx that i would like to integrate within nextcloud. 
With community containers is there a way to have bind mounted volumes from unraid (ie outside of /var/lib/docker)?
I'm quite confused on this subject. For example i see  jellyfin or plex containers (i wonder why one would installs these inside nextcloud but that's not the subject), i can't imagine to have my plex library on a volume in /var/lib/docker as it's located on a ssd pool.
Could we have for example?:
> "volumes": [
                {
                    "source": "/mnt/user/appdata/paperless",
                    "destination": "/usr/src/paperless/data",
                    "writeable": true
                },
                {
                    "source": "/mnt/user/paperless/media", // or should the nextcloud datadir be mounted here? What is the media folder for?
                    "destination": "/usr/src/paperless/media",
                    "writeable": true
                }/* Not 100% sure on the volumes below:,
                {
                    "source": "/mnt/user/paperless/export", // should this be included in the backup solution?
                    "destination": "/usr/src/paperless/export",
                    "writeable": true
                },
                {
                    "source": "/mnt/user/paperless/consume", // should this be Nextclouds datadir or can it be ommitted?
                    "destination": "/usr/src/paperless/consume",
                    "writeable": true // does this needs to be writeable?
                }*/
            ],
            "backup_volumes": [
                "nextcloud_aio_paperless_ngx_data" // and more?
            ]

What should be these folders permissions and owner ? They need to be writable by paperless and paperless manages permissions for the different users.
Paperless (as nextcloud) supports oauth so i guess it would be easier if nextcloud user and paperless user match.
/usr/src/paperless/data is for config of paperless
/usr/src/paperless/media is for the files and as paperless supports multi user you can have /usr/src/paperless/media/$user. So i imagine you could mount it at the same level as documents
/usr/src/paperless/consume is for dropping files to be processed by paperless (uploaded by a user or a network scanner). It needs to be writable. In paperless there several options for using this folder.
